### ISSUE TITLE
New Varactor Library for ngspice

### ISSUE THREAD

**Issue by ra3xdh** [ra3xdh/qucs_s#759]
### Discussed in https://github.com/ra3xdh/qucs_s/discussions/758

<div type='discussions-op-text'>

<sup>Originally posted by **tomhajjar** June 16, 2024</sup>
Attached is a new library for ngspice called Varactor_RF.lib. Also attached is a project used to test the models and develop the library, Models include packaging parasitics. Should be used with the latest ngspice build that allows larger Vj and M.

![2024-06-15_185048](https://github.com/ra3xdh/qucs_s/assets/102838604/ae9d8133-1742-433a-9fcd-afccaf92b315)

![2024-06-15_185516](https://github.com/ra3xdh/qucs_s/assets/102838604/ff1bfa6d-1cfc-4fb3-bcf1-e1cbf60a652c)

![2024-06-15_214751](https://github.com/ra3xdh/qucs_s/assets/102838604/f1b1bc6e-6d63-43ea-80a8-8da1ff202285)

[Varactor_ngspice_prj.zip](https://github.com/user-attachments/files/15855870/Varactor_ngspice_prj.zip)

</div>

**Comment by tomhajjar** [comment]
The library doesn't include the BB814 dual varactor. I couldn't get a test fixture to work for the two diode in series. Single diode and two in parallel works but not in series. Capacitance should be 1/2 of one diode but it measured negative...

![2024-06-15_214751](https://github.com/ra3xdh/qucs_s/assets/102838604/f31b0e7d-3575-425d-adc6-b023e52f330e)

![2024-06-16_082134](https://github.com/ra3xdh/qucs_s/assets/102838604/ebf5761a-b87b-4fe1-913f-5e3fb29b26d5)

![2024-06-16_082047](https://github.com/ra3xdh/qucs_s/assets/102838604/e343ab42-f038-44eb-a1d5-b51486f0f670)

**Comment by tomhajjar** [comment]
I got the BB814 dual varactor to simulate using S-parameters instead of the previous method. Unsure why the original method doesn't work for a dual varactor. 

Updated Varactor_RF.lib so please delete previous one in zip above.

![2024-06-16_204930](https://github.com/ra3xdh/qucs_s/assets/102838604/0761c470-0c14-4c02-8a59-178d45fda3ce)

[Varactor_ngspice_prj.zip](https://github.com/user-attachments/files/15860456/Varactor_ngspice_prj.zip)

**Comment by ra3xdh** [comment]
Varactor_RF library has been uploaded by #767. I have slightly modified the library source, so it works now with both Qucsator and Ngspice.

**Comment by tomhajjar** [comment]
I am almost done updating Varactor.lib. Attached sample is 95% done.

I added descriptions and changed Rs to reflect the real varactor. This library is OK as long as one recognizes it only works at maybe VHF frequencies and 95% of the diodes are obsolete. 

[Varactor.zip](https://github.com/user-attachments/files/15888807/Varactor.zip)

**Comment by ra3xdh** [comment]
> I am almost done updating Varactor.lib. Attached sample is 95% done.

Thanks! Please open a new PR when you finish the library update. I will upload it in the repository.

**Comment by tomhajjar** [comment]
Here is the finished library. 

[Varactor.zip](https://github.com/user-attachments/files/15889213/Varactor.zip)

**Comment by ra3xdh** [comment]
> Here is the finished library.

I have added the updated `Varactor.lib` library to #767.
